<a href="https://colab.research.google.com/github/carterliston013/DCS-110/blob/main/Intro_to_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print('Hello World!')

Hello World!


#Variables
In this section, we'll discuss variables


In [ ]:
a = 'Boston'

This established the variable 'a' as 'Boston'

In [ ]:
print (a)

Boston


This command prints the variable 'a'.

In [ ]:
print (a[-1])

n


This command prints the last letter of the variable 'a'.

In [ ]:
2+3

5

This command is the command for adding 2 and 3.

In [ ]:
b = 2
c = 3

This command establishes the variable b as 2 and c as 3.

In [ ]:
b+c

5

This uses the addition function to add the variables b and c.

In [ ]:
d = b + c

This establishes the variable d as the sum of variables b and c.

In [ ]:
b=2
c=3
d=b+c
print(d)

5


This is all commands in the same window showing it is able to be put into the same command.
It also includes the print function for the variable d.

In [ ]:
type(d)

int

This is showing that the data type for variable d is an integer (int).

In [ ]:
type(a)

str

This is showing that the variable type is string (str), meaning words.

In [ ]:
5**2

25

This is the command for a power function (**).

In [ ]:
print(5**2)

25


This is the same command as above, using the print function as well.

# Data Structures (Containers)

List

In [ ]:
My_List=[1, 2, 3, 5, 'a']

This command establishes the values in 'My List'

In [ ]:
print(My_List)

[1, 2, 3, 5, 'a']


This command uses the print function to print the values in 'My List'

In [ ]:
print(My_List[0])

1


This uses the [0] to print the first value in 'My List'.

In [ ]:
My_List[0]=10

This command reestablishes the first value in My_List as 10, instead of 1.

In [ ]:
print(My_List)

[10, 2, 3, 5, 'a']


This command reprints the list as it has been changed by the previous command.

Tuples

In [ ]:
My_Tuple=(12, 'finger', 7, 'ichabod', 'porcupine', 19)

This command establishes 'My Tuple' and its values, which allows both numbers and words as data types.

In [ ]:
print(My_Tuple)

(12, 'finger', 7, 'ichabod', 'porcupine', 19)


This prints the Tuple as it has been established.

In [ ]:
print(My_Tuple[0])

12


This prints the first value of 'My Tuple' using [0]

In [ ]:
My_Tuple[0]=38

TypeError: 'tuple' object does not support item assignment

This error shows that you are not able to change the values that have been set into the Tuple.

Dictionary

In [ ]:
My_Dictionary={
    'Name':"Cringle",
    'Age':348,
    'Gender':'Male',
    'City':'North Pole'

}

This establishes the fields and values into My_Dictionary.

In [ ]:
My_Dictionary['Name']

'Cringle'

This function pulls the value associated with 'Name' in My_Dictionary.

Sets

In [ ]:
My_Set={1, 2, 3, 4, 5, 5}

This command establishes the values put into My_Set.

In [ ]:
print(My_Set)

{1, 2, 3, 4, 5}


This command shows that, when you print a set, the values that are duplicated are only shown once.

In [ ]:
#We cannot access values in the set
My_Set[0]

TypeError: 'set' object is not subscriptable

This shows that you cannot pull the first value in a set because sets are unordered and not retrievable by position.

# **Functions**

In [ ]:
temp = (4 * 9/5) + 32
print(temp)

39.2


This command shows the creation and completion of a function.

In [ ]:
temp = (-6 * 9/5) + 32
print(temp)

21.2


This is exemplifies the same function as the previous command with different numbers.

In [2]:
def celsius_to_fahrenheit (celsius):
  fahrenheit = (celsius * 9/5) + 32
  return fahrenheit

This defines a Celsius to Fahrenheit calculation function with Celsius as the main variable and Fahrenheit as the returned solution value.

In [ ]:
celsius_to_fahrenheit (100)

212.0

This uses the established Celsius to Fahrenheit function to return the Fahrenheit solution (212) from the Celsius variable (100).

Formatted String Literals

In [ ]:
print(f'The temperature now is {celsius_to_fahrenheit(472)}F')

The temperature now is 881.6F


This prints a sentence, while including a functional result by beginning with the f function allowing a calculation to be done in part with the sentence.

Loops

In [ ]:
My_Temperature_List = [0, 11, 28, 42, 69]

This command establishes the values of My_Temperature_List

In [3]:

celsius_to_fahrenheit(11)

51.8

This shows the calculated temperature in Fahrenheit when the Celsius variable is 11.

In [ ]:
for celsius in My_Temperature_List:
  print (celsius_to_fahrenheit(celsius))

32.0
51.8
82.4
107.6
156.2


This command inputs the values of My_Temperature_List into the Celsius to Fahrenheit function as the variable Celsius to return the Fahrenheit values associated with the values in the list.

# BMI Function

In [ ]:
def BMI_Calc (height, weight):
  BMI = (weight/ height**2) * 703
  return BMI

This is a function defined for calculating BMI using weight in lbs and height in inches.

In [ ]:
BMI_Calc (73, 180)

23.74554325389379

This uses the values 73 (for height in inches) and 180 (weight in lbs) to return the solution of the BMI for these values.

In [ ]:
My_BMI_List = [(73, 180), (60, 250), (82,100)]

This establishes values for the (height, weight) input into a list called My_BMI_List.

In [ ]:
for (height, weight)in My_BMI_List:
  print(BMI_Calc(height, weight))

23.74554325389379
48.81944444444445
10.455086258179655


This command uses the values from My_BMI_List to input them into the BMI_Calc as (height, weight) and return their BMI values respectively.

In [ ]:
BMI=18
if BMI<20:
  print('Underweight')
else:
  print('Normal')

Underweight


This establishes ranges for text returns according to BMI values, BMI<20 being Underweight and all other values as normal.

Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

These commands import libraries to be references using shorter codes.

In [ ]:
sum_c = sum(My_Temperature_List)
print(c)

3


This command equalizes the sum of c and the sum of My_Temperature_List as the same, making the sum of c the same value.

In [ ]:
len_c = len(My_Temperature_List)
print(len_c)

5


The len function sums the number of objects in a list, summing the number of objects in My_Temperature_List and equalizing that to len_c.

In [ ]:
a = np.mean(My_Temperature_List)
print(a)

30.0


This references the numpy library allowing for the use of the mean function. This averages the values in My_Temperature_List.